presto-orc/src/main/java/com/facebook/presto/orc/reader/MapDirectSelectiveStreamReader.java

```diff
@@ -0,0 +1,638 @@
+                continue;
+            }
+
+            assert outputPositions[i] == nextPosition;
+
+            offsets[positionIndex + 1] = offsets[i + 1] - nestedSkipped;
+            for (int j = offsets[positionIndex]; j < offsets[positionIndex + 1]; j++) {
+                nestedOutputPositions[nestedOutputPositionCount] = nestedOutputPositions[nestedOutputPositionCount + nestedSkipped];
+                nestedOutputPositionCount++;
+            }
+
+            if (compactNulls) {
+                nulls[positionIndex] = nulls[i];
+            }
+            outputPositions[positionIndex] = nextPosition;
+
+            positionIndex++;
+            if (positionIndex >= positionCount) {
+                break;
+            }
+            nextPosition = positions[positionIndex];
+        }
+
+        outputPositionCount = positionCount;
+    }
+
+    private BlockLease newLease(Block block, BlockLease...fieldBlockLeases)
+    {
+        valuesInUse = true;
+        return ClosingBlockLease.newLease(block, () -> {
+            for (BlockLease lease : fieldBlockLeases) {
+                lease.close();
+            }
+            valuesInUse = false;
+        });
+    }
+
+    @Override
+    public void throwAnyError(int[] positions, int positionCount)
+    {
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .addValue(streamDescriptor)
+                .toString();
+    }
+
+    @Override
+    public void close()
+    {
+        systemMemoryContext.close();
+    }
+
+    @Override
+    public void startStripe(InputStreamSources dictionaryStreamSources, List<ColumnEncoding> encoding)
+            throws IOException
+    {
+        presentStreamSource = missingStreamSource(BooleanInputStream.class);
+        lengthStreamSource = missingStreamSource(LongInputStream.class);
+
+        readOffset = 0;
+        nestedReadOffset = 0;
+
+        presentStream = null;
+        lengthStream = null;
+
+        rowGroupOpen = false;
+
+        keyReader.startStripe(dictionaryStreamSources, encoding);
+        valueReader.startStripe(dictionaryStreamSources, encoding);
+    }
+
+    @Override
+    public void startRowGroup(InputStreamSources dataStreamSources)
+            throws IOException
+    {
+        presentStreamSource = dataStreamSources.getInputStreamSource(streamDescriptor, PRESENT, BooleanInputStream.class);
+        lengthStreamSource = dataStreamSources.getInputStreamSource(streamDescriptor, LENGTH, LongInputStream.class);
+
+        readOffset = 0;
+        nestedReadOffset = 0;
+
+        presentStream = null;
+        lengthStream = null;
+
+        rowGroupOpen = false;
+
+        keyReader.startRowGroup(dataStreamSources);
+        valueReader.startRowGroup(dataStreamSources);
+    }
+
+    @Override
+    public long getRetainedSizeInBytes()
+    {
+        return INSTANCE_SIZE + sizeOf(outputPositions) + sizeOf(offsets) + sizeOf(nulls) +
```

Comment:
nit: more readable if it is one param per line

Reply:
Sure. Changing to 

```
        return INSTANCE_SIZE +
                keyStreamReader.getRetainedSizeInBytes() + 
                valueStreamReader.getRetainedSizeInBytes();
```